### ISSUE TITLE
Regression with least-squares loss

### ISSUE THREAD

**Issue by stephanegaiffas** [pyensemble/wildwood#4]
Code a `ForestRegressor` class

- Code a Base `BaseForest` class to avoid code duplication between `ForestClassifier` and `ForestRegressor`. We won't inherit from sklearn's `MultiOutputMixin` since we won't handle multioutput (multi-label) problems for now
- Code the base `ForestRegressor` with correct attributes (no more `n_classes` etc)
- First thing is to code the criterions in `_impurity` module, we can follow what is done in `https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/tree/_criterion.pyx`
- The criterion are passed as functions, so genericity and code reuse should be easy

**Comment by stephanegaiffas** [comment]
Remains to finish _tree and _tree_context and should be fine

**Comment by stephanegaiffas** [comment]
Solved in #14
